src/Env/WordPressEnvBridge.php

```diff
@@ -584,9 +586,10 @@ public function dumpCached(string $file): bool
             }
 
             // For env loaded from file, dump the variable definition.
-            $content .= "putenv('{$key}={$slashed}');\n";
-            $content .= "\$_ENV['{$key}'] = '{$slashed}';\n";
-            (strpos($key, 'HTTP_') !== 0) and $content .= "\$_SERVER['{$key}'] = '{$slashed}';\n\n";
+            $content .= "putenv('{$key}={$preparedValue}');\n";
+            $content .= "\$_ENV['{$key}'] = '{$preparedValue}';\n";
+            (strpos($key, 'HTTP_') !== 0)
+                and $content .= "\$_SERVER['{$key}'] = '{$preparedValue}';\n\n";
```

Comment:
This ss a different behavior than before. 
Before the environment was never storing the "filtered" value. That was used only for PHP constant.
And the reason is that environemt is always expected to be a string, while filtered value could be a integer, a float, a boolean, etc.

By storing the prepared value in environment, you are going to change the original value. See https://3v4l.org/pfaoh

Reply:
Thanks a lot for the explanation 🙏
So the situation is that for an env variable like: `AUTH_COOKIE=BAD&Auth<Cookie`
the generated cache is:
```PHP
define('AUTH_COOKIE', 'BAD&amp;Auth');
putenv('AUTH_COOKIE=BAD&Auth<Cookie');
$_ENV['AUTH_COOKIE'] = 'BAD&Auth<Cookie';
$_SERVER['AUTH_COOKIE'] = 'BAD&Auth<Cookie';
```
in which the values are not consistent, First question is that I'm not sure if this is the expected? My assumption was the all values should be the same in the generated cache.

To get around this, does something like this on line 569 makes sense?
```PHP
$preparedValue = $value !== $filtered && is_scalar($filtered)
    ? is_string($filtered) ? $filtered : var_export($filtered, true)
    : $slashed;
```

Reply:
As you can see, the issue is only for the constant `define`.

So you need to change nothing in the lines I was was mentioning in the first comment. Because those lines refers to `putenv`, `$_ENV`, and `$_SERVER` which do not need fixing.

So all your changes up to like 579 are fine, just leave anything blow that line alone, and it'll be fine.

Reply:
The [commit](https://github.com/wecodemore/wpstarter/pull/136/commits/78846d2c186d4413a7338f8f161c1cd86adce791) introducing this change reverted since the only real other changes before line 579 was using `addslashes` instead of `str_replace` which is out of scope of this PR.